File: android-room-sample-master/app/src/javaDagger/java/com/emreeran/android/roomsample/di/DbModule.java
package com.emreeran.android.roomsample.di;

import android.app.Application;
import android.arch.persistence.room.Room;

import com.emreeran.android.roomsample.db.SampleDb;
import com.emreeran.android.roomsample.db.dao.CommentDao;
import com.emreeran.android.roomsample.db.dao.FeedDao;
import com.emreeran.android.roomsample.db.dao.LikeDao;
import com.emreeran.android.roomsample.db.dao.PostDao;
import com.emreeran.android.roomsample.db.dao.UserDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Emre Eran on 26.05.2018.
 */
@Module
public class DbModule {

    @Singleton
    @Provides
    public SampleDb provideDb(Application application) {
        return Room.databaseBuilder(application, SampleDb.class, "sample.db").build();
    }

    @Singleton
    @Provides
    public UserDao provideUserDao(SampleDb db) {
        return db.userDao();
    }

    @Singleton
    @Provides
    public PostDao providePostDao(SampleDb db) {
        return db.postDao();
    }

    @Singleton
    @Provides
    public CommentDao provideCommentDao(SampleDb db) {
        return db.commentDao();
    }

    @Singleton
    @Provides
    public LikeDao provideLikeDao(SampleDb db) {
        return db.likeDao();
    }

    @Singleton
    @Provides
    public FeedDao provideFeedDao(SampleDb db) {
        return db.feedDao();
    }
}

File: android-room-sample-master/app/src/kotlinDagger/java/com/emreeran/android/roomsample/di/DbModule.kt
package com.emreeran.android.roomsample.di

import android.app.Application
import android.arch.persistence.room.Room
import com.emreeran.android.roomsample.db.SampleDb
import com.emreeran.android.roomsample.db.dao.*
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 *  Created by Emre Eran on 20.04.2018.
 */
@Module
class DbModule {
    private val _dbName = "sample.db"

    @Singleton
    @Provides
    fun provideDb(app: Application): SampleDb {
        return Room.databaseBuilder(app, SampleDb::class.java, _dbName).build()
    }

    @Singleton
    @Provides
    fun provideUserDao(db: SampleDb): UserDao {
        return db.userDao()
    }

    @Singleton
    @Provides
    fun providePostDao(db: SampleDb): PostDao {
        return db.postDao()
    }

    @Singleton
    @Provides
    fun provideCommentDao(db: SampleDb): CommentDao {
        return db.commentDao()
    }

    @Singleton
    @Provides
    fun provideLikeDao(db: SampleDb): LikeDao {
        return db.likeDao()
    }

    @Singleton
    @Provides
    fun provideFeedDao(db: SampleDb): FeedDao {
        return db.feedDao()
    }
}
